sdk/log/exporter.go

```diff
@@ -221,12 +221,31 @@ func (e *bufferExporter) enqueue(ctx context.Context, records []Record, rCh chan
 // EnqueueExport enqueues an export of records in the context of ctx to be
 // performed asynchronously. This will return true if the exported is
 // successfully enqueued, false otherwise.
+//
+// The passed records are held after this call returns.
 func (e *bufferExporter) EnqueueExport(records []Record) bool {
 	if len(records) == 0 {
 		// Nothing to enqueue, do not waste input space.
 		return true
 	}
-	return e.enqueue(context.Background(), records, nil) == nil
+
+	data := exportData{ctx: context.Background(), records: records}
+
+	e.inputMu.Lock()
+	defer e.inputMu.Unlock()
+
+	// Check stopped before enqueueing now that e.inputMu is held. This
+	// prevents sends on a closed chan when Shutdown is called concurrently.
+	if e.stopped.Load() {
+		return false
```

Comment:
```suggestion
		return true
```

Otherwise we will unnecessarily "loop" until context is done in ForceFlush which can be forever if user passes `context.Background()`.